src/vw-serving/src/vw_serving/model_manager.py

```diff
@@ -0,0 +1,270 @@
+import os
+import boto3
+import redis
+import time
+import signal
+import json
+import logging
+import subprocess
+import shutil
+import multiprocessing
+from multiprocessing import Process
+from pathlib import Path
+
+from vw_serving.utils import dynamic_import, parse_s3_url, gen_random_string
+import vw_serving.sagemaker.config.environment as environment
+from vw_serving.sagemaker import integration as integ
+from vw_serving.firehose_producer import FirehoseProducer
+from vw_serving.sagemaker.exceptions import convert_to_algorithm_error, raise_with_traceback, AlgorithmError, CustomerError
+from vw_serving.serve import REDIS_PUBLISHER_CHANNEL
+from boto3.dynamodb.conditions import Key
+
+
+logger = logging.getLogger()
+
+
+class ModelDBClient(object):
+    def __init__(self, table_session):
+        self.table_session = table_session
+
+    def check_model_record_exists(self, experiment_id, model_id):
+        if self.get_model_record(experiment_id, model_id) is None:
+            return False
+        else:
+            return True
+
+    def get_model_record(self, experiment_id, model_id):
+        response = self.table_session.query(
+            ConsistentRead=True,
+            KeyConditionExpression=Key('experiment_id').eq(experiment_id) & Key('model_id').eq(model_id)
+        )
+        for i in response['Items']:
+            return i
+        return None
+
+
+class ExperimentDBClient(object):
+    def __init__(self, table_session):
+        self.table_session = table_session
+
+    def get_experiment_record(self, experiment_id):
+        response = self.table_session.query(
+            ConsistentRead=True,
+            KeyConditionExpression=Key('experiment_id').eq(experiment_id)
+        )
+        for i in response['Items']:
+            return i
+        return None
+
+
+class ModelManager():
+    def __init__(self):
+        self.sagemaker_tar_gz = self._check_sagemaker_model_dir()
+        self.experiment_id = os.getenv(environment.EXPERIMENT_ID, "default_experiment")
+        self.model_id = os.getenv(environment.MODEL_ID, "default_model")
+
+        self.poll_db = os.getenv(environment.MODEL_METADATA_POLLING, 'true').lower() == 'true'
```

Comment:
poll_db decides whether to do soft deployment updates or not. Can you make this false by default?

Reply:
Done